### PR TITLE
[CCR] Auto follow pattern APIs adjustments

### PR DESCRIPTION
**Pull request by martijnvg**
* Changed the resource id of auto follow patterns to be a user defined name in all auto follow pattern crud APIs.
* Which leader cluster to use is now specified in the body of an auto follow pattern.
* Fail when an unfollowed leader index matches with two or more auto follow patterns. The unfollow leader index will then not be followed and the error can be seen in the auto follow stats API. The reason this validation is needed is, because multiple auto follow patters can be defined for a single leader cluster.

### CODE REVIEW

**Comment by martijnvg** [comment]
run gradle build tests

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/AutoFollowIT.java [hunk 1]
```diff
@@ -44,7 +46,12 @@ public void testAutoFollow() throws Exception {
         createLeaderIndex("logs-201812", leaderIndexSettings);
 
         // Enabling auto following:
-        putAutoFollowPatterns("logs-*", "transactions-*");
+        if (randomBoolean()) {
+            putAutoFollowPatterns("my-pattern", new String[] {"logs-*", "transactions-*"});
+        } else {
+            putAutoFollowPatterns("my-pattern1", new String[] {"logs-*"});
```
Comment (bleskes) [inline x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/AutoFollowIT.java]: 💯

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/AutoFollowIT.java [hunk 2]
```diff
@@ -172,9 +180,54 @@ public void testAutoFollowParameterAreDelegated() throws Exception {
         });
     }
 
-    private void putAutoFollowPatterns(String... patterns) {
+    public void testConflictingPatterns() throws Exception {
```
Comment (bleskes) [inline x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/AutoFollowIT.java]: if two patterns match an index, why is that a problem? we used to be able to do it before no?

Comment (martijnvg) [reply x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/AutoFollowIT.java]: So only one auto follow pattern would trigger the index following and when applying the second one would result in an error (same follow index already exists in follow cluster). Also which auto follow pattern gets applied first is undefined.

Comment (bleskes) [reply x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/AutoFollowIT.java]: I looked at it more and I realized I got confused - with the new model we can still specify multiple patterns in a single auto follow config (is this a better name?). All good. Thanks for explaining.

Comment (martijnvg) [reply x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/AutoFollowIT.java]: > with the new model we can still specify multiple patterns in a single auto follow config (is this a better name?)

Yes, we can still do that.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**